@here/harp-mapview/lib/text/TextStyleCache.ts

```diff
@@ -127,3 +148,371 @@ export class TextLayoutStyleCache {
+                styledTextRenderer !== undefined ? styledTextRenderer.poiRenderer : undefined;
+        }
+        if (this.m_defaultStyle.textCanvas === undefined) {
+            if (this.m_defaultStyle.fontCatalog !== undefined) {
+                logger.warn(
+                    `FontCatalog '${this.m_defaultStyle.fontCatalog}' set in TextStyle '${
+                        this.m_defaultStyle.name
+                    }' not found, using default fontCatalog(${
+                        defaultTextCanvas!.fontCatalog.name
+                    }).`
+                );
+            }
+            this.m_defaultStyle.textCanvas = defaultTextCanvas;
+            this.m_defaultStyle.poiRenderer = defaultPoiRenderer;
+        }
+
+        // Initialize theme text styles.
+        this.m_theme.textStyles!.forEach(element => {
+            this.m_textStyles.set(
+                element.name!,
+                this.createTextElementStyle(element, element.name!)
+            );
+        });
+        // tslint:disable-next-line:no-unused-variable
+        for (const [, style] of this.m_textStyles) {
+            if (style.textCanvas === undefined) {
+                if (style.fontCatalog !== undefined) {
+                    const styledTextRenderer = textRenderers.find(
+                        textRenderer => textRenderer.fontCatalog === style.fontCatalog
+                    );
+                    style.textCanvas =
+                        styledTextRenderer !== undefined
+                            ? styledTextRenderer.textCanvas
+                            : undefined;
+                    style.poiRenderer =
+                        styledTextRenderer !== undefined
+                            ? styledTextRenderer.poiRenderer
+                            : undefined;
+                }
+                if (style.textCanvas === undefined) {
+                    if (style.fontCatalog !== undefined) {
+                        logger.warn(
+                            `FontCatalog '${style.fontCatalog}' set in TextStyle '${
+                                style.name
+                            }' not found, using default fontCatalog(${
+                                defaultTextCanvas!.fontCatalog.name
+                            }).`
+                        );
+                    }
+                    style.textCanvas = defaultTextCanvas;
+                    style.poiRenderer = defaultPoiRenderer;
+                }
+            }
+        }
+    }
+
+    /**
+     * Retrieves a [[TextElementStyle]] for [[Theme]]'s [[TextStyle]] id.
+     */
+    getTextElementStyle(styleId?: string): TextElementStyle {
+        let result;
+        if (styleId === undefined) {
+            result = this.m_defaultStyle;
+        } else {
+            result = this.m_textStyles.get(styleId);
+            if (result === undefined) {
+                result = this.m_defaultStyle;
+            }
+        }
+        return result;
+    }
+
+    /**
```

Comment:
These two functions below getRenderStyle / getLayoutStyle were duplicated in PoiManager and TileGeometryCreator. I've moved them here.